### PR TITLE
cpu_model_negative:fix error msg change

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 2034937

Signed-off-by: nanliu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
(1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: STARTED
| (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (1.85 s)
 (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: STARTED
\ (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: PASS (1.80 s)
 (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: STARTED
- (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: PASS (1.85 s)

**Comment by nanliu-r** [comment]
@PaulYuuu Please help to review this patch, thanks.

File: qemu/tests/cfg/cpu_model_negative.cfg [hunk 1]
```diff
@@ -10,6 +11,9 @@
         - smp_lt_maxcpu:
             wrong_cmd = "-smp 8,maxcpus=4,cores=2,threads=2,dies=1,sockets=2"
             warning_msg = 'maxcpus must be equal to or greater than smp'
+                required_qemu = [, 6.2.0)
+            warning_msg = 'qemu-kvm: Invalid CPU topology: product of the hierarchy must match maxcpus'
+                required_qemu = [6.2, )
```
Comment (PaulYuuu) [inline qemu/tests/cfg/cpu_model_negative.cfg:16]: Syntax error.

Comment (nanliu-r) [reply qemu/tests/cfg/cpu_model_negative.cfg]: > Syntax error.

update and thanks.

File: qemu/tests/cfg/cpu_model_negative.cfg [hunk 2]
```diff
@@ -10,6 +11,9 @@
         - smp_lt_maxcpu:
             wrong_cmd = "-smp 8,maxcpus=4,cores=2,threads=2,dies=1,sockets=2"
             warning_msg = 'maxcpus must be equal to or greater than smp'
+                required_qemu = [, 6.2.0)
+            warning_msg = 'qemu-kvm: Invalid CPU topology: product of the hierarchy must match maxcpus'
```
Comment (PaulYuuu) [inline qemu/tests/cfg/cpu_model_negative.cfg:15]: `warning_msg` is not a valid variant, so indent is not applied, the second `warning_msg` and `required_qemu` will overwrite the first one.

Comment (nanliu-r) [reply qemu/tests/cfg/cpu_model_negative.cfg]: > `warning_msg` is not a valid variant, so indent is not applied, the second `warning_msg` and `required_qemu` will overwrite the first one.

update, and thanks.

**Comment by nanliu-r** [comment]
(1/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: STARTED
| (1/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (1.91 s)
 (2/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.old_msg.q35: STARTED
\ (2/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.old_msg.q35: CANCEL: Got host qemu version:6.2.0-10, which is not in [, 6.2.0) (1.75 s)
 (3/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.new_msg.q35: STARTED
- (3/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.new_msg.q35: PASS (1.84 s)
 (4/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: STARTED
/ (4/4) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: PASS (1.90 s)

@PaulYuuu please help to review this again. Thanks.

File: qemu/tests/cfg/cpu_model_negative.cfg [hunk 3]
```diff
@@ -9,7 +10,13 @@
             warning_msg = ".${enforce_flag}' not found"
         - smp_lt_maxcpu:
             wrong_cmd = "-smp 8,maxcpus=4,cores=2,threads=2,dies=1,sockets=2"
-            warning_msg = 'maxcpus must be equal to or greater than smp'
+            variants:
+                - old_msg:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/cpu_model_negative.cfg:14]: Hum, how to say. After this part is added, always one test case will be skipped, but they are the same scenario.
So, do we have other solutions to fix the warning_msg? e.g:
```
Host_RHEL.x.x.x:
    warning_msg = 'maxcpus must be equal to or greater than smp'
Host_RHEL.y.y.y:
    warning_msg = 'qemu-kvm: Invalid CPU topology: product of the hierarchy must match maxcpus'
```
or
```
cfg:
    old_warning_msg = 'maxcpus must be equal to or greater than smp'
    new_warning_msg = 'qemu-kvm: Invalid CPU topology: product of the hierarchy must match maxcpus'
---
py:
        qemu_version = utils_qemu.get_qemu_version(qemu_binary)[0]
        required_qemu_version = self.params["required_qemu_version"]
        if qemu_version in VersionInterval('[, 6.2.0)'):
            warning_msg = old_warning_msg
        else:
            warning_msg = new_warning_msg
```

Comment (nanliu-r) [reply qemu/tests/cfg/cpu_model_negative.cfg]: @PaulYuuu thanks and update, please help to review.

**Comment by nanliu-r** [comment]
(1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: STARTED
| (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (1.88 s)
 (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: STARTED
\ (2/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: PASS (1.84 s)
 (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: STARTED
- (3/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: PASS (1.89 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**